src/LearnMe.Core/Services/Calendar/GoogleCalendar.cs

```diff
@@ -0,0 +1,159 @@
+
+        public GoogleCalendar(IGoogleAPIconnection googleAPIconnection,
+                              ICrudRepository<CalendarEvent> repository,
+                              IMapper mapper,
+                              ILogger<GoogleCalendar> logger)
+        {
+            var token = googleAPIconnection.GetToken();
+            _calendarService = googleAPIconnection.CreateCalendarService(token, _applicationName);
+            _repository = repository;
+            _mapper = mapper;
+            _logger = logger;
+        }
+
+        //TODO Add Event builder implementation
+        public async Task<bool> CreateEventAsync(CalendarEventDto eventData,
+                                                 string calendarId = "primary",
+                                                 string timezone = "Europe/Warsaw",
+                                                 bool isRecurringEvent = false)
+        {
+            CalendarEvent newDbEvent = _mapper.Map<CalendarEvent>(eventData);
+
+            Event newCalendarEvent = new Event()
+            {
+                Summary = newDbEvent.Title,
+                Description = newDbEvent.Description
+            };
+
+            DateTime? startDateTime = newDbEvent.Start;
+            EventDateTime start = new EventDateTime()
+            {
+                DateTime = startDateTime,
+                TimeZone = timezone
+            };
+            newCalendarEvent.Start = start;
+
+            DateTime? endDateTime = newDbEvent.End;
+            EventDateTime end = new EventDateTime()
+            {
+                DateTime = endDateTime,
+                TimeZone = timezone
+            };
+            newCalendarEvent.End = end;
+
+            if (isRecurringEvent)
+            {
+                String[] recurrence = new String[] { "RRULE:FREQ=DAILY;COUNT=7" };
+                newCalendarEvent.Recurrence = recurrence.ToList();
+            }
+
+            // FOR FUTURE - ADD ATTENDEES
+            //EventAttendee[] attendees = new EventAttendee[]
+            //{
+            //    new EventAttendee() { Email = "<---email-address--->@gmail.com" },
+            //};
+            //newEvent.Attendees = attendees.ToList();
+
+            var createdEvent = await _calendarService.Events.Insert(newCalendarEvent, calendarId).ExecuteAsync();
+            newDbEvent.CalendarId = createdEvent.Id;
+
+            return await _repository.InsertAsync(newDbEvent);
+        }
+
+        public async Task<bool> DeleteEventAsync(int id)
+        {
+            return await _repository.DeleteAsync(id);
+        }
+
+        private void SynchronizeDatabaseWithCalendar(string calendarId = "primary")
+        {
+            EventsResource.ListRequest request = _calendarService.Events.List(calendarId);
+            IEnumerable<Event> eventsFromCalendarResult = request.ExecuteAsync().Result.Items;
+
+            var allDatabaseEvents = _repository.GetAllAsync().Result;
```

Comment:
Result out

Reply:
Ogólnie z Resultem jest taki kejs, że Result powoduje coś takiego jak `Task Starvation`, doczytaj o co w tym chodzi. W skrócie, cała apka stoi, ty dokładasz na task schedulera kolejne taski i mimo to, że zwiększasz wydajność serwera (więcej threadów, więcej pamięci) to i tak to nic nie da, bo Result zatrzymuje apke na czas wykonania